### PR TITLE
feat: expose selected/yanked files as API

### PR DESCRIPTION
**Pull request by sxyazi**
Prepare for https://github.com/sxyazi/yazi/pull/646 - @Rolv-Apneseth

### CODE REVIEW

**Comment by sxyazi** [comment]
Yeah it's already, I added a new `cx.yanked` and `cx.active.selected` property, it should simplify the code at https://github.com/sxyazi/yazi/pull/646.

Let me know if I missed something needed/better to have!

File: yazi-core/src/manager/commands/open.rs [hunk 1]
```diff
@@ -98,13 +107,30 @@ impl Manager {
 		});
 	}
 
-	fn quit_with_selected(selected: &[&File]) -> bool {
+	fn guess_its_folder(&self, url: &Url) -> bool {
```
Comment (Rolv-Apneseth) [inline yazi-core/src/manager/commands/open.rs:110]: Maybe just `guess_folder`?

**Comment by Rolv-Apneseth** [comment]
With these changes, when in visual mode, yanking/cutting only does the file under the cursor.

**Comment by sxyazi** [comment]
> With these changes, when in visual mode, yanking/cutting only does the file under the cursor

Yeah this is a known issue, I will fix it in the next PR. This PR focuses on the plugin interface

File: yazi-core/src/manager/commands/open.rs (hunk 1 above)
Comment (sxyazi) [reply yazi-core/src/manager/commands/open.rs]: Applied this suggestion, thanks!

**Comment by Rolv-Apneseth** [comment]
Ah ok. All seems good to me then

**Comment by Rolv-Apneseth** [comment]
I'm gonna be away from this computer for a day or two so I'm gonna just push the changes to #646 and rebase after this is merged

**Pull request merged**